### PR TITLE
GitHub Actions add e2e test

### PR DESCRIPTION
**Pull request by imechoim**

### CODE REVIEW

File: .github/workflows/e2e_tests.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          path: datalens-ui
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: 'npm'
+          cache-dependency-path: ./datalens-ui/package-lock.json
+
+      - name: Run Docker Compose
+        working-directory: ./datalens
+        run: docker compose up -d
```
Comment (vrozaev) [inline .github/workflows/e2e_tests.yml:30]: We should change code here. This code starts hardcoded datalens image from current docker-compose.yml:

```
  datalens:
    container_name: datalens-ui
    image: ghcr.io/datalens-tech/datalens-ui:0.795.0
```

We need to change this file like this:

```
  datalens:
    container_name: datalens-ui
    dockerfile: ./datalens-ui/Dockerfile
```

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+version: "3.7"
+
+services:
+  datalens:
+    build:
+      context: ../datalens-ui
+      dockerfile: ../datalens-ui/Dockerfile
+    image: result/latest
```
Comment (vrozaev) [inline docker-compose.yml:8]: Should we move this file to /tests directore?

File: tests/playwright.config.ts [hunk 3]
```diff
@@ -68,7 +71,7 @@ const playwrightConfig: PlaywrightTestConfig<DatalensTestFixtures> = {
         },
         headless: !headful,
         baseURL,
-        storageState: 'artifacts/storageState.json',
+        // storageState: 'artifacts/storageState.json',
```
Comment (vrozaev) [inline tests/playwright.config.ts:74]: If we don't need it anymore than we can just remove it.

File: tests/suites/collections/OpenCollectionsPage.test.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import {Page} from '@playwright/test';
+
+import {openTestPage} from '../../utils';
+import datalensTest from '../../utils/playwright/globalTestDefinition';
+
+datalensTest.describe('Collections page', () => {
+    datalensTest('Open page', async ({page}: {page: Page}) => {
+        await openTestPage(page, `/collections`);
+    });
+});
```
Comment (vrozaev) [inline tests/suites/collections/OpenCollectionsPage.test.tsx:10]: Can we be sure that this test will not be run together with inner e2e tests?

File: tests/utils/playwright/datalens/datalens-test-setup.ts [hunk 5]
```diff
@@ -5,19 +5,17 @@ import {FullConfig, Page, chromium} from '@playwright/test';
 
 import {AuthRobotSettings} from '../types';
 
-import {authenticate} from './utils';
```
Comment (vrozaev) [inline tests/utils/playwright/datalens/datalens-test-setup.ts:8]: Should we remove the authenticate from utils as well?

File: .github/workflows/e2e_tests.yml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        working-directory: ./datalens
+        run: docker compose -f ./docker-compose.yml -f ../datalens-ui/docker-compose.yml up -d & cd ../datalens-ui && npm ci && npm run test:install:chromium & wait
+
+
+      # - name: Install dependencies Datalens-UI 
+      #   working-directory: ./datalens-ui
+      #   run: npm ci && npm run test:install:chromium
+      - name: Run Test 
+        working-directory: ./datalens-ui
+        run: E2E_DOMAIN="http://localhost:8080" E2E_TEST_NAME_PATTERN="Open page" npm run test:e2e
```
Comment (vrozaev) [inline .github/workflows/e2e_tests.yml:38]: Now we start just one test. We need to start all *opensource* tests. We can:

1. Put them to separate folder and grep by folder name
2. Use tags https://playwright.dev/docs/test-annotations#tag-tests
3. Name opensource tests with different prefix and grep tests by test name.

File: tests/utils/playwright/datalens/datalens-test-setup.ts (hunk 5 above)
Comment (vrozaev) [reply tests/utils/playwright/datalens/datalens-test-setup.ts:8]: upd: no, other tests are depends on that.

File: tests/playwright.config.ts (hunk 3 above)
Comment (vrozaev) [reply tests/playwright.config.ts:74]: upd: no, other tests are depends on that.

File: tests/utils/playwright/datalens/datalens-test-setup.ts [hunk 7]
```diff
@@ -44,15 +42,7 @@ export async function datalensTestSetup({config, afterAuth, authSettings}: Datal
...
-        await authenticate({
-            page,
-            baseUrl,
-            passportUrl: authSettings.url,
-            login: authSettings.login,
-            password: authSettings.password,
-            afterAuth,
-            retryCount: AUTH_RETRY,
-        });
+        await afterAuth?.({page});
```
Comment (Flunt1k) [inline tests/utils/playwright/datalens/datalens-test-setup.ts:45]: Should we remove afterAuth as well? It seems redundant without authentication

File: tests/utils/playwright/datalens/datalens-test-setup.ts [hunk 8]
```diff
@@ -44,15 +42,7 @@ export async function datalensTestSetup({config, afterAuth, authSettings}: Datal
     }
 
     try {
```
Comment (Flunt1k) [inline tests/utils/playwright/datalens/datalens-test-setup.ts:44]: Should we remove afterAuth as well with all try/catch? It seems redundant without authentication

File: tests/utils/playwright/datalens/datalens-test-setup.ts (hunk 5 above)
Comment (Flunt1k) [reply tests/utils/playwright/datalens/datalens-test-setup.ts:8]: I would only suggest creating a separate playwright config for public tests

**Review by vrozaev — APPROVED** [review]

**Pull request merged**